src/charm.py

```diff
@@ -121,92 +123,107 @@ def _get_database_relation_info(self) -> dict:
             "endpoints": relation_data["endpoints"],
         }
 
-    def _update_layer(self) -> None:
-        """Updates the Pebble configuration layer and config if changed."""
-        self.unit.status = MaintenanceStatus("Applying pebble layer")
-
-        if not self._container.get_plan().to_dict():
-            # Push the config on first layer update to avoid PathError
-            self._container.push(self._hydra_config_path, self._config, make_dirs=True)
-            logger.info("Pushed hydra config")
-
-        self._container.add_layer(self._container_name, self._hydra_layer, combine=True)
-        logger.info("Pebble plan updated with new configuration, replanning")
-
-        try:
-            self._container.replan()
-        except ChangeError as err:
-            logger.error(str(err))
-            self.unit.status = BlockedStatus("Failed to replan")
-            return
-
+    def _run_sql_migration(self) -> None:
+        """Runs a command to create SQL schemas and apply migration plans."""
+        process = self._container.exec(
+            ["hydra", "migrate", "sql", "-e", "--config", self._hydra_config_path, "--yes"],
+            timeout=20.0,
+        )
         try:
-            current_config = self._container.pull(self._hydra_config_path).read()
-        except (ProtocolError, PathError) as err:
-            logger.error(str(err))
-            self.unit.status = BlockedStatus(str(err))
-        else:
-            if current_config != self._config:
-                self._container.push(self._hydra_config_path, self._config, make_dirs=True)
-                logger.info("Updated hydra config, restarting the container")
-                self._container.restart()
-
-    def _update_container(self, event) -> None:
-        """Update configs, pebble layer and run database migration."""
-        self.unit.status = MaintenanceStatus("Configuring hydra container")
+            stdout, _ = process.wait_output()
+            logger.info(f"Executing automigration: {stdout}")
+        except ExecError as err:
+            logger.error(f"Exited with code {err.exit_code}. Stderr: {err.stderr}")
+            self.unit.status = BlockedStatus("Database migration job failed")
```

Comment:
What happens if this times out? 

We need to also have an action for running migrations, maybe in case of timeout (or all errors?) we could print a message telling the user to run the migration action

Reply:
The migration shouldn't really take that long, but I removed the 20s timeout. Errors should be caught by `ops.pebble.ExecError`. I added an action to run the migration manually.

Reply:
How long the migration takes depends on the volume of the data in the database and on the migration commands. If the migration runs from a hook we should have a timeout set, if the user runs the migration we can wait it out.